Kitodo-API/src/main/java/org/kitodo/serviceloader/KitodoServiceLoader.java

```diff
@@ -52,20 +74,242 @@ public KitodoServiceLoader(Class clazz, String modulePath) {
 
     /**
      * Loads a module from the classpath which implements the constructed clazz.
-     * 
+     * Frontend files of all modules will be loaded into the core module.
+     *
      * @return A module with type T.
      */
     @SuppressWarnings("unchecked")
     public T loadModule() {
 
         loadModulesIntoClasspath();
+        loadBeans();
+        loadFrontendFilesIntoCore();
 
         ServiceLoader<T> loader = ServiceLoader.load(clazz);
 
         return loader.iterator().next();
     }
 
     /**
+     * Loads bean classes and registers them to the FacesContext. Afterwards they can be used in all
+     * frontend files
+     */
+    private void loadBeans() {
+        try {
+
+            Path moduleFolder = FileSystems.getDefault().getPath(modulePath);
+            DirectoryStream<Path> stream = Files.newDirectoryStream(moduleFolder, "*.jar");
+
+            for (Path f : stream) {
+                JarFile jarFile = new JarFile(f.toString());
```

Comment:
Class `JarFile` implements the `Closeable` / `Autocloseable` interface which means that we can use the try-with-resources-statement here. This stream object is not closed in current implementation which is an other good reason to change it.